v2/tools/generator/internal/codegen/pipeline/prune_resources_with_lifecycle_owned_by_parent.go

```diff
@@ -69,19 +75,65 @@ func PruneResourcesWithLifecycleOwnedByParent(configuration *config.Configuratio
 	return stage
 }
 
+func checkPrunedEmptyProperties(defs astmodel.TypeDefinitionSet, emptyPrunedProps []astmodel.TypeName) (astmodel.TypeDefinitionSet, error) {
+	emptyObjectVisitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType: tagEmptyObjectARMProperty,
+	}.Build()
+
+	for _, emptyPrunedProp := range emptyPrunedProps {
+		// we need to add the noConversion tag on ARM type for the empty pruned property to relax the validation for convertToARM function.
+		armDef, err := GetARMTypeDefinition(defs, emptyPrunedProp)
+		if err != nil {
+			return nil, err
+		}
+
+		tempResult := make(astmodel.TypeDefinitionSet)
+		for _, def := range defs {
+			updatedDef, err := emptyObjectVisitor.VisitDefinition(def, armDef.Name())
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to visit definition %s", def.Name())
+			}
+			tempResult.Add(updatedDef)
+		}
+		defs = tempResult
```

Comment:
Visiting a type is relatively expensive - a visitor walks the entire passed type, making precise edits where required. It would be substantially more efficient to visit each type exactly once.

Reply:
Yep, makes sense. I've refactored the code